### DISCUSSION TITLE
How to: Use esphome as RS485 to Wifi bridge

### DISCUSSION THREAD

**Discussion by DerTiger1** [omerfaruk-aran/esphome_samsung_hvac_bus#66] [General]
It is possible to bridge the serial connection over Wifi to a virtual COM-Port and to use the Samsung S-Net software for remote controlling your HVAC Unit:

1.  Add [esphome-stream-server](https://github.com/oxan/esphome-stream-server) to your esphome yaml:
    external_components:
      - source: github://oxan/esphome-stream-server
    stream_server:
2. Compile and upload
3. Install a virtual COM-port driver (e.g.: [https://www.hw-group.com/software/hw-vsp3-virtual-serial-port](https://www.hw-group.com/software/hw-vsp3-virtual-serial-port))
4. Start the virtual COM-Port and make sure that "NVT enabled" is unchecked under settings and make sure to set the port to 6638 (which is default of oxan/esphome-stream-server) 
5. Connect with the Samsung S-Net Software using the option F1/F2 (to change COM-Port you have to click on the "hearbeat" icon in the upper left)
  S-Net NASA (new protocol) systems: [SNET-Pro2](https://www.samsunghvac.com/Software-Downloads/SNET-Pro2)
  S-Net non NASA systems: [SNET-Pro](https://www.samsunghvac.com/Software-Downloads/SNET-Pro)

Edit: Sometimes it takes time until the connection is established - disconnect and re-connect helps.
Don't try anything until you see a window like this:
![grafik](https://github.com/lanwin/esphome_samsung_ac/assets/95141154/51501436-166c-4e48-a2a6-5021153d31e7)

Enjoy

**Comment by lanwin** [comment]
Thank you.

**Comment by bzumik1** [comment]
@DerTiger1 that is great to know, thanks a lot!

**Comment by lanwin** [comment]
Do you see a way to log the data that is written thru the virtual COM port?

**Comment by DerTiger1** [comment]
> Do you see a way to log the data that is written thru the virtual COM port?

A long time ago we were using serial sniffers for this but I can't find any software that is free to use.
As the packets should be the RAW RS485 data I managed to record them using wireshark with a filter on the specified TCP Port of the serial to wifi connection.
